packages/router/src/route_definition/content.rs

```diff
@@ -0,0 +1,167 @@
+use std::{collections::BTreeMap, fmt::Debug};
+
+use dioxus::prelude::*;
+
+use crate::navigation::NavigationTarget;
+
+/// The content of a [`Route`] or [`ParameterRoute`].
+///
+/// [`Route`]: super::Route
+/// [`ParameterRoute`]: super::ParameterRoute
+#[derive(Clone)]
+pub enum RouteContent {
+    /// No content.
+    ///
+    /// Can be used to make a route transparent and have its nested routes be nested one level less
+    /// deep.
+    Empty,
+    /// A single component.
+    Component(Component),
```

Comment:
I'll try that. I hate the `as Component` thing.

Reply:
I don't think this is possible. I did some basic testing, trying to store a `Box<dyn Fn(Scope) -> Element + 'static>` as content. While creating the `Box` works fine, `cx.provide_context()` doesn't accept it, because it doesn't implement `Clone`. Creating a trait requiring `Fn(Scope) -> Element` and `Clone` doesn't work either, because that trait can't be made into an object. Any other ideas?

Reply:
For clarity: this matters because the router also needs to clone the components, to tell the `Outlet`s which ones are active. 

While that could be changed to an index, then the components themselves would need to be in the context, which doesn't work, as explained above.

And I don't think `as Component` is bad enough to warrant an `RC` or `Arc`.

Reply:
I ran into a similar problem with OwnedAttributeValue and a boxed Any. [dyn-clone](https://docs.rs/dyn-clone/latest/dyn_clone/trait.DynClone.html) allows you to clone trait objects.

Or it could accept &'static dyn Fn(Scope) -> Element which would be very cheaply cloneable because it is a pointer. Something like [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=109794ddd984f29793a131b036f59527)

Reply:
Alright, I'll try that tomorrow

Reply:
Sorry for the long wait, life got in the way. Just tried your suggestion and storing the content as `&'static dyn Fn(Scope) -> Element` works just fine. The problem is that the `rsx!` macro doesn't accept that.

This is the message if I try to pass it in directly:
```
mismatched types
expected fn pointer `fn(dioxus::prelude::Scope<'_, _>) -> Option<dioxus::prelude::VNode<'_>>`
    found reference `&dyn for<'r> std::ops::Fn(dioxus::prelude::Scope<'r>) -> Option<dioxus::prelude::VNode<'_>>`
```

And this if I try to dereferece it (X is the variable holding the content):
```
expected struct, variant or union type, found local variable `X`
not a struct, variant or union type
```

The first message seems to imply that the macro can only accept `fn(Scope) -> Element`. If that is correct, is there a workaround we could use? Or would this need to be changed outside the router to work?